Sources/KlaviyoSwiftExtension/KlaviyoExtension.swift

```diff
@@ -24,6 +30,28 @@ public enum KlaviyoExtensionSDK {
         bestAttemptContent: UNMutableNotificationContent,
         contentHandler: @escaping (UNNotificationContent) -> Void,
         fallbackMediaType: String = "jpeg") {
+        // handle badge setting from the push notification payload
+        if let badgeConfig = bestAttemptContent.userInfo["badge_config"] as? String {
+            guard let appGroup = Bundle.main.object(forInfoDictionaryKey: "klaviyo_app_group") as? String,
+                  let userDefaults = UserDefaults(suiteName: appGroup) else {
+                return
+            }
+
+            var newBadgeValue: Int?
+            switch badgeConfig {
+            case KlaviyoBadgeConfig.incrementOne.rawValue:
+                let currentBadgeCount = userDefaults.integer(forKey: "badgeCount")
+                newBadgeValue = currentBadgeCount + 1
+            default:
+                if let badgeValue = bestAttemptContent.userInfo["badge_value"] as? Int {
+                    newBadgeValue = badgeValue
+                }
+            }
```

Comment:
```suggestion
        if let badgeConfigValue = bestAttemptContent.userInfo["badge_config"] as? String {
            guard let appGroup = Bundle.main.object(forInfoDictionaryKey: "klaviyo_app_group") as? String,
                  let userDefaults = UserDefaults(suiteName: appGroup) else {
                return
            }
            
            guard let badgeConfig = KlaviyoBadgeConfig(rawValue: badgeConfigValue) else { return }

            var newBadgeValue: Int?
            switch badgeConfig {
            case .incrementOne:
                let currentBadgeCount = userDefaults.integer(forKey: "badgeCount")
                newBadgeValue = currentBadgeCount + 1
            case .setCount, .setProperty:
                if let badgeValue = bestAttemptContent.userInfo["badge_value"] as? Int {
                    newBadgeValue = badgeValue
                }
            }
```

I think this is a cleaner way of handling the conversion from a raw value to an enum case. It also allows us to avoid the `default` case, which [in general we should try to avoid](https://www.swiftbysundell.com/tips/avoiding-default-cases-in-switch-statements/)

Reply:
Thanks @ab1470 I was looking for ways to avoid the raw value and this is neat. On the `default` case point, I agree in the case of apps but since we are a SDK, I wonder if there is still value in defining what the default behaviour should be in the case we add newer option to this on the backend?

Reply:
ah I see. The question is what behavior do we want in case a new, unknown-to-the-SDK value is supplied by the backend. With my suggested code above, it'll hit line 40, won't be able to convert the raw value to an enum case, and then return without doing anything because it's an unrecognized value. Do we want to do that? Or do we want to try to handle the unknown value anyway?

Reply:
So yeah the question is whether we're comfortable letting the `default` case handle any unknown future values. The advantage of making sure that we're always doing _something_ is also a risk. Suppose someday we add a backend config value like `"do_nothing"`, but the SDK doesn't know how to explicitly handle it so it gets handled by the `default` case and increments the badge anyway

Reply:
ok, I'll leave it up to you then, as you have more knowledge on badges stuff than I do. That being said, we can still add a little safety and readability by adding an `.unknown` case to the enum and creating a custom initializer:
```swift
private enum KlaviyoBadgeConfig {
    case incrementOne
    case setCount
    case setProperty
    case unknown(value: String)
    
    init(rawValue: String) {
        switch rawValue {
        case "increment_one": self = .incrementOne
        case "set_count": self = .setCount
        case "set_property": self = .setProperty
        default: self = .unknown(value: rawValue)
        }
    }
}
```

and then use it like this:
```swift
let badgeConfig = KlaviyoBadgeConfig(rawValue: badgeConfigValue)

var newBadgeValue: Int?

switch badgeConfig {
case .incrementOne:
    let currentBadgeCount = userDefaults.integer(forKey: "badgeCount")
    newBadgeValue = currentBadgeCount + 1
case .setCount, .setProperty, .unknown:
    if let badgeValue = bestAttemptContent.userInfo["badge_value"] as? Int {
        newBadgeValue = badgeValue
    }
```

Reply:
based on [this](https://klaviyo.slack.com/archives/C080HQERMJP/p1734722960414899?thread_ts=1734721301.288619&cid=C080HQERMJP) thread we will do nothing for the .unknown case